### PR TITLE
Changes for creating a single feed service

### PR DESCRIPTION
**Pull request by shamnadps**

### CODE REVIEW

File: backend/.env [hunk 1]
```diff
@@ -11,3 +11,4 @@ YOUTUBE_CHANNEL_LIST_LIMIT=20
 FACEBOOK_PAGE_TOKEN=EAADVvvpW6goBAIt6Q9ZAiUcTOqCvFQWaxHKcvSTxlZCdaTYC5m3qBtGT8k25fnUKQQmYeZBJnZAC8mw77iRcCDZCpLDfFauCTtwe9L5aelzf56ABotZA4EBdZAR3yHMcT5ErUwScz0TaYmZBlCdADHWu1aXJ9pzYbMzSJgUugIje6Xu5KTaQ9rz6PvQcMPKZBmBcZD
 VANTAA_FACEBOOK_PAGE=vantaaservices
 FACEBOOK_GRAPH_API=https://graph.facebook.com/v3.1/
+NUMBER_OF_FEEDS=20
```
[the remark this answers is not in the document]
Comment (shamnadps) [reply backend/.env]: Yes. thats correct.

File: backend/package.json [hunk 2]
```diff
@@ -32,6 +32,7 @@
     "pg": "^7.4.3",
     "reflect-metadata": "^0.1.12",
     "rss-parser": "^3.4.3",
+    "rss-parser-browser": "^2.2.2",
```
[the remark this answers is not in the document]
Comment (shamnadps) [reply backend/package.json]: This is a rss parsing plugin which can fetch custom fields in the rss response.

File: backend/src/event/event.module.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { Module, Logger } from '@nestjs/common';
```
[the remark this answers is not in the document]
Comment (shamnadps) [reply backend/src/event/event.module.ts]: We have different feeds. Rss, Events, facebook, youtube, twitter and instagram at the moment.
The feeds which have similar model are handled together and the ones which are not are done in separate modules.

File: backend/src/event/event.service.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+export class EventFeedService {
+
+    constructor(
+        private readonly feedService: FeedService,
+        private readonly logger: Logger) {
+        this.logger = new Logger('EventFeedService');
+    }
+
+    async onModuleInit() {
+        await this.fetchEventFeeds();
```
[the remark this answers is not in the document]
Comment (shamnadps) [reply backend/src/event/event.service.ts]: Yes thats true. I will remove that. I just write the async/await everywhere just got used to it. Will correct it everywhere.

File: backend/src/event/event.service.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
+import { Injectable, Logger } from '@nestjs/common';
+import * as converter from 'json-style-converter/es5';
+import sources from './event.sources';
+import { FeedService } from '../feeds/feed.service';
+const Parser = require('rss-parser-browser');
+
+@Injectable()
+export class EventFeedService {
```
[the remark this answers is not in the document]
Comment (shamnadps) [reply backend/src/event/event.service.ts]: Feed is generalized model. Event feed is to handle only event feeds.

File: backend/src/facebook-feed/facebook-feed.service.ts [hunk 6]
```diff
@@ -22,14 +18,15 @@ export class FacebookFeedService {
         await this.fetchFacebookFeeds();
     }
 
-    @Transaction()
     async fetchFacebookFeeds() {
         try {
+            this.logger.log('Fetching facebook feeds Started');
             const vantaaFacebookDataUrl = `${config.facebookGraphApi}${sources.vantaa}/posts?`
```
[the remark this answers is not in the document]
Comment (shamnadps) [reply backend/src/facebook-feed/facebook-feed.service.ts]: What do you mean?

File: backend/src/feeds/feed.dto.ts [hunk 7]
```diff
@@ -0,0 +1,17 @@
+export class FeedDto {
+    id: number;
+    author: string;
+    author_thumbnail: string;
```
[the remark this answers is not in the document]
Comment (shamnadps) [reply backend/src/feeds/feed.dto.ts]: Yes. This was the model shared by Maria and I have asked for comments on the channel about the naming.

File: backend/src/feeds/feed.service.ts [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    async getFeeds(type: string, limit: number) {
+        try {
+            const feeds = type ? await this.getFeedsByType(type, limit) : await this.getAll(limit);
+            return feeds.sort((a, b) => compareDesc(a.pub_date, b.pub_date)).slice(0, limit);
+        } catch (error) {
+            this.logger.error(`Failed to get feeds: ${error}`)
+        }
+    }
+
+    async getAll(limit: number) {
```
[the remark this answers is not in the document]
Comment (shamnadps) [reply backend/src/feeds/feed.service.ts]: Yes I will add that.

File: backend/src/rss-feed/rss-feed.service.ts [hunk 9]
```diff
@@ -24,32 +21,35 @@ export class RssFeedService {
...
+        this.logger.log('Fetching Rss feeds Started');
         try {
             await Promise.all([
-                parser.parseURL(sources.news).then(this.addSourceName('rss_news')).then(this.persistIntoDb),
-                parser.parseURL(sources.stories).then(this.addSourceName('rss_stories')).then(this.persistIntoDb),
-                parser.parseURL(sources.aikuisopisto).then(this.addSourceName('rss_aikuisopisto')).then(this.persistIntoDb),
-                parser.parseURL(sources.nuorten).then(this.addSourceName('rss_nuorten')).then(this.persistIntoDb),
-                parser.parseURL(sources.kaupunginmuseo).then(this.addSourceName('rss_kaupunginmuseo')).then(this.persistIntoDb),
-                parser.parseURL(sources.events).then(this.addSourceName('rss_events')).then(this.persistIntoDb)
+                parser.parseURL(sources.news).then(this.addSourceName('Rss_News')).then(this.persistIntoDb),
```
[the remark this answers is not in the document]
Comment (shamnadps) [reply backend/src/rss-feed/rss-feed.service.ts]: Yes I will add that

File: backend/src/rss-feed/rss-feed.service.ts [hunk 10]
```diff
@@ -24,32 +21,35 @@ export class RssFeedService {
...
+                parser.parseURL(sources.stories).then(this.addSourceName('Rss_Stories')).then(this.persistIntoDb),
+                parser.parseURL(sources.aikuisopisto).then(this.addSourceName('Rss_Aikuisopisto')).then(this.persistIntoDb),
+                parser.parseURL(sources.nuorten).then(this.addSourceName('Rss_Nuorten')).then(this.persistIntoDb),
+                parser.parseURL(sources.kaupunginmuseo).then(this.addSourceName('Rss_Kaupunginmuseo')).then(this.persistIntoDb),
             ]);
         } catch (error) {
```
[the remark this answers is not in the document]
Comment (shamnadps) [reply backend/src/rss-feed/rss-feed.service.ts]: Yes agree. I will update.

File: backend/src/event/event.module.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (StormBlade97) [reply backend/src/event/event.module.ts]: ok,  i see now

File: backend/src/feeds/feed.dto.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (StormBlade97) [reply backend/src/feeds/feed.dto.ts]: i see. just code style, nothing too important

File: backend/src/rss-feed/rss-feed.service.ts (hunk 10 above)
[the remark this answers is not in the document]
Comment (StormBlade97) [reply backend/src/rss-feed/rss-feed.service.ts]: in that case i think you can drop Promise.all and try catch entirely. Just a bunch of promise minding their own business

**Comment by StormBlade97** [comment]
Answered to your answer.
If you are incrementally refactoring something, add a @TODO in the code so that I know you are working on it

**Pull request merged**